fdbrpc/include/fdbrpc/LoadBalance.actor.h

```diff
@@ -480,16 +498,23 @@ struct RequestData : NonCopyable {
 	                                      int requiredReplicas) {
 		if (model && (compareReplicas || FLOW_KNOBS->ENABLE_REPLICA_CONSISTENCY_CHECK_ON_READS)) {
 			ASSERT(requestStream != nullptr);
-			int requiredReplicaCount =
-			    compareReplicas ? requiredReplicas : FLOW_KNOBS->CONSISTENCY_CHECK_REQUIRED_REPLICAS;
-			return replicaComparison(request,
-			                         response,
-			                         requestStream->getEndpoint().token.first(),
-			                         alternatives,
-			                         channel,
-			                         requiredReplicaCount);
+			if (FLOW_KNOBS->ENABLE_REPLICA_CONSISTENCY_CHECK_ON_READS) {
+				return replicaComparison(request,
+				                         response,
+				                         requestStream->getEndpoint().token.first(),
+				                         alternatives,
+				                         channel,
+				                         FLOW_KNOBS->READ_CONSISTENCY_CHECK_REQUIRED_REPLICAS);
+			}
+			if (compareReplicas) {
```

Comment:
It seems this must be true here, thus no need to check.

Reply:
Will solve it in https://github.com/apple/foundationdb/pull/12164 at first